cmd/src/batch_common.go

```diff
@@ -295,19 +295,32 @@ func executeBatchSpec(ctx context.Context, opts executeBatchSpecOpts) error {
 	}
 	batchCompletePending(pending, fmt.Sprintf("Found %d workspaces with steps to execute", len(tasks)))
 
-	execOpts := executor.Opts{
+	// EXECUTION OF TASKS
+
+	svc.InitCache(opts.flags.cacheDir)
+	svc.InitExecutor(ctx, executor.NewExecutorOpts{
 		CacheDir:      opts.flags.cacheDir,
-		ClearCache:    opts.flags.clearCache,
 		CleanArchives: opts.flags.cleanArchives,
 		Creator:       workspaceCreator,
 		Parallelism:   opts.flags.parallelism,
 		Timeout:       opts.flags.timeout,
 		KeepLogs:      opts.flags.keepLogs,
 		TempDir:       opts.flags.tempDir,
+	})
+
+	pending = batchCreatePending(opts.out, "Checking cache for changeset specs")
+	uncachedTasks, cachedSpecs, err := svc.CheckCache(ctx, tasks, opts.flags.clearCache)
+	if err != nil {
+		return err
+	}
+	if len(uncachedTasks) > 0 {
+		batchCompletePending(pending, fmt.Sprintf("Found %d cached changeset specs. %d tasks need to be executed", len(cachedSpecs), len(uncachedTasks)))
```

Comment:
I think we might have hit the point where we need a pluralisation helper, since we have things like "Found 1 cached changeset specs" coming out of this. (It doesn't look like we have one in the `src-cli` tree right now.)

To _completely_ nitpick, it might read better if this is a semicolon instead of a full stop between the clauses, since our other outputs aren't full sentences.

Reply:
Done! And yes, pluralisation helper would be nice.

Reply:
(Also: semicolon nitpicks are amazing. I always want to use semicolons but never dare to, because I'm not sure whether it's correct to use)